### DISCUSSION TITLE
Error Loading data to TimeBase using pyspark

### DISCUSSION THREAD

**Discussion by ankitsoin** [finos/TimeBase-CE#217] [Q&A]
Hello,

May I know the most recommended way of loading data to timebase from avro file of around 500mb in size?
File is not present on TimeBase server.

I was trying to load data using PySpark from Avro file (400-500 MB) by simply iterating data frame and sending message to the fixed stream using below code and getting error 
"TypeError: can't pickle SwigPyObject objects".

Code:
#building connection to timebase and db.open(False)
#reading data from file into pyspark sorted dataframe: df

stream = db.getStream(StreamKey)

options = tbapi.LoadingOptions()
options.writeMode = tbapi.WriteMode('APPEND')
loader = stream.createLoader(options)

mObj = tbapi.InstrumentMessage()
mObj.TypeName = 'path.to.class.ClassName' 

#ClassName class extends InstrumentMessage, timestamp and symbol were passed to parent class. Also contains assignment of other #fields to class attributes like price and exchange_name 

def save(x):
	  mObj.symbol = x["sym"]
	  mObj.timestamp = x["timestamp"]
	  mObj.price = x["price"]
	  mObj.exchange_name = x["exchange_name"]
	  loader.send(mObj)
	
df.foreach(lambda x: save(x))

#closing loader and db connection

Could you please help us on above issue or any other workaround to load data?
Thanks

**Comment by alex-karpovich** [comment]
I suppose you should check object type of the properties of "x[]"-  it should be primitive types. 
Or conver before assign to msg. Like 
mObj.price = toDouble( x["price"])

**Reply by ankitsoin** [reply]
Thanks for replying,
I have tried using same but it didnt work

I beleive workers in spark cant connect with outside world thats why it is causing exception.
So i have used spark continous streams and it worked using foreach on streaming dataframe.

**Reply by alex-karpovich** [reply]
So you are able to load data into TimeBase stream now?

**Reply by ankitsoin** [reply]
Yes, I am able to load.
